File: core/shell/list_container_proxy.cc
// Copyright 2025 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.
#include "core/public/list_container_proxy.h"
namespace lynx {
namespace shell {
void ListContainerProxy::ScrollByListContainer(int32_t tag, float offset_x,
                                               float offset_y, float original_x,
                                               float original_y) {
  if (list_engine_proxy_) {
    list_engine_proxy_->ScrollByListContainer(tag, offset_x, offset_y,
                                              original_x, original_y);
  }
}

void ListContainerProxy::ScrollToPosition(int32_t tag, int index, float offset,
                                          int align, bool smooth) {
  if (list_engine_proxy_) {
    list_engine_proxy_->ScrollToPosition(tag, index, offset, align, smooth);
  }
}

void ListContainerProxy::ScrollStopped(int32_t tag) {
  if (list_engine_proxy_) {
    list_engine_proxy_->ScrollStopped(tag);
  }
}

}  // namespace shell

}  // namespace lynx

File: platform/android/lynx_android/src/main/java/com/lynx/tasm/behavior/ui/list/container/ListContainerProxy.java
// Copyright 2025 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.
package com.lynx.tasm.behavior.ui.list.container;

public class ListContainerProxy {
  private UIListContainer mUIList;
  private long mNativeEnginePtr;
  private long mNativeContainerPtr;

  private ListContainerProxy() {}

  public ListContainerProxy(UIListContainer uiList, long nativeEnginePtr) {
    mUIList = uiList;
    mNativeEnginePtr = nativeEnginePtr;
    mNativeContainerPtr = nativeCreate(mNativeEnginePtr);
  }

  public void scrollByListContainer(int sign, float x, float y, float originalX, float originalY) {
    if (mNativeContainerPtr != 0) {
      nativeScrollByListContainer(mNativeContainerPtr, sign, x, y, originalX, originalY);
    }
  }

  public void scrollToPosition(int sign, int position, float offset, int align, boolean smooth) {
    if (mNativeContainerPtr != 0) {
      nativeScrollToPosition(mNativeContainerPtr, sign, position, offset, align, smooth);
    }
  }

  public void scrollStopped(int sign) {
    if (mNativeContainerPtr != 0) {
      nativeScrollStopped(mNativeContainerPtr, sign);
    }
  }

  public void destroy() {
    if (mNativeContainerPtr != 0) {
      nativeDestroy(mNativeContainerPtr);
      mNativeContainerPtr = 0;
    }
  }

  // ------------------JNI Method start -----------------------

  // create 'list_container_proxy' C++ object
  private native long nativeCreate(long enginePtr);

  private native void nativeScrollByListContainer(
      long ptr, int sign, float x, float y, float originalX, float originalY);

  private native void nativeScrollToPosition(
      long ptr, int sign, int position, float offset, int align, boolean smooth);

  private native void nativeScrollStopped(long ptr, int sign);

  // release 'list_container_proxy' C++ object
  private native void nativeDestroy(long containerPtr);

  // ------------------JNI Method end -----------------------
}
